tests/bluetooth/bsim_bt/bsim_test_l2cap/src/main_l2cap_ecred.c

```diff
@@ -0,0 +1,563 @@
+		FAIL("Received out of sequence data.");
+	}
+
+	int retval = memcmp(buf->data + sizeof(int), chan->payload + sizeof(int), buf->len - sizeof(int));
+
+	if (retval) {
+		FAIL("Payload received didn't match expected value memcmp returned %i", retval);
+	}
+
+	if (chan->chan_id == 0) {
+		/*By the time we rx on chan 0, we should have alrady received on chan1*/
+		if ((channels[1].sdus_received != (channels[0].sdus_received + 1))) {
+			FAIL("Didn't receive on channel 1 first: channels[0].sdus_received:%i channels[1].sdus_received:%i",
+			     channels[0].sdus_received,
+			     channels[1].sdus_received);
+		}
+	}
+
+	chan->sdus_received++;
+
+	return 0;
+}
+
+static void chan_sent_cb(struct bt_l2cap_chan *l2cap_chan)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	chan->buf = 0;
+
+	LOG_DBG("chan_id: %d", chan->chan_id);
+}
+
+static volatile int num_connect_chans;
+static void chan_connected_cb(struct bt_l2cap_chan *l2cap_chan)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	LOG_DBG("chan_id: %d", chan->chan_id);
+
+	LOG_DBG("tx.mtu %d, tx.mps: %d, rx.mtu: %d, rx.mps %d", sys_cpu_to_le16(chan->le.tx.mtu),
+		sys_cpu_to_le16(chan->le.tx.mps), sys_cpu_to_le16(chan->le.rx.mtu),
+		sys_cpu_to_le16(chan->le.rx.mps));
+
+	num_connect_chans++;
+}
+
+static void chan_disconnected_cb(struct bt_l2cap_chan *l2cap_chan)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	LOG_DBG("chan_id: %d", chan->chan_id);
+
+	chan->in_use = false;
+	num_connect_chans--;
+}
+
+static void chan_status_cb(struct bt_l2cap_chan *l2cap_chan, atomic_t *status)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	LOG_DBG("chan_id: %d, status: %ld", chan->chan_id, *status);
+}
+
+static void chan_released_cb(struct bt_l2cap_chan *l2cap_chan)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	LOG_DBG("chan_id: %d", chan->chan_id);
+}
+
+static void chan_reconfigured_cb(struct bt_l2cap_chan *l2cap_chan)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	LOG_DBG("chan_id: %d", chan->chan_id);
+}
+
+static const struct bt_l2cap_chan_ops l2cap_ops = {
+	.alloc_buf = chan_alloc_buf_cb,
+	.recv = chan_recv_cb,
+	.sent = chan_sent_cb,
+	.connected = chan_connected_cb,
+	.disconnected = chan_disconnected_cb,
+	.status = chan_status_cb,
+	.released = chan_released_cb,
+	.reconfigured = chan_reconfigured_cb,
+};
+
+static void connect_num_channels(uint8_t num_l2cap_channels)
+{
+	struct channel *chan = NULL;
+
+	#define L2CAP_ECRED_CHAN_MAX 5
+	struct bt_l2cap_chan *allocated_channels[L2CAP_ECRED_CHAN_MAX] = { NULL };
+	#undef L2CAP_ECRED_CHAN_MAX
+	int err = 0;
+
+	for (int i = 0; i < num_l2cap_channels; i++) {
+		chan = get_free_channel();
+		if (!chan) {
+			FAIL("failed, chan not free");
+			return;
+		}
+		chan->le.chan.ops = &l2cap_ops;
+		chan->le.rx.mtu = DATA_MTU;
+		chan->le.rx.mps = DATA_MPS;
```

Comment:
Given that the ops, MTU and MPS are always set to the same value, and in multiple places, we could consider moving them into the `get_free_channel` where we already initialize other values, or perhaps add a new function that initializes the channel. 

Seems odd to initialize some things in `get_free_channel`, and some other things in the caller, when it could be done in a single place.